src/RequestExecutor/WsRequestExecutor.ts

```diff
@@ -0,0 +1,64 @@
+import { RequestExecutor } from './RequestExecutor';
+import { Response } from './Response';
+import { Request } from './Request';
+import { Protocol } from '../Handlers';
+import { RequestExecutorOptions } from './HttpRequestExecutor';
+import { inject, injectable } from 'tsyringe';
+import WebSocket, { ErrorEvent, MessageEvent } from 'ws';
+
+@injectable()
+export class WsRequestExecutor implements RequestExecutor {
+  public protocol: Protocol = Protocol.WS;
+
+  constructor(
+    @inject(RequestExecutorOptions)
+    private readonly options: RequestExecutorOptions
+  ) {}
+
+  public async execute(options: Request): Promise<Response> {
+    const client = new WebSocket(options.url, options.headers);
+
+    client.onopen = () => {
+      client.send(options.body);
+    };
+
+    const result = await Promise.race<Response>([
```

Comment:
You should listen to `close` event. `error` and `close` are not the same. E.g. server can close a socket by some reason with ws closing code, e.g `1010` (see rfc6455).